### PR TITLE
[Barcode Scanner] Improve barcode scanner reading accuracy

### LINKED ISSUE #12656: [Barcode Scanner] Improve barcode scanner reading accuracy
There have been issues reported p1724846800296649-slack-C013AAPA4G0 regarding barcode scanners reading inaccuracy when scanning barcodes.

The goal of the task is to research and come up with potential solutions on improving the barcode scanners reading accuracy.

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #12656 

### Description

There have been issues reported p1724846800296649-slack-C013AAPA4G0 regarding barcode scanners reading inaccuracy when scanning barcodes.

In this PR, I address an issue where the target resolution for barcode scanning was not being set correctly. The width and height of the preview view were always 0 because we attempted to access these dimensions before the layout had been calculated.

The target resolution is essential for ensuring accurate barcode detection. By fixing the timing of when the target resolution is set, we now correctly capture the preview dimensions after the layout has been established.

Through testing, this change has shown noticeable improvements in barcode reading accuracy, especially in scenarios where the scanner previously struggled to detect barcodes.

### Steps to reproduce

**Before this PR**
1. Navigate to either orders list, products, or order creation screen
2. Click on the barcode icon
3. Try scanning the barcode from different angles except pointing camera straight to barcode (angled right, left, from far away ...etc)
4. See that the scanner reads different value (most of the time) than what is in the barcode resulting in failing to find the product

**After this PR**
1. Navigate to either orders list, products, or order creation screen
2. Click on the barcode icon
3. Try scanning the barcode from different angles except pointing camera straight to barcode (angled right, left, from far away ...etc)
4. See that the scanner reads correct value (most of the time) than what is in the barcode resulting in fetching the product successfully.

### Testing information

Try scanning the barcode in various challenging ways! Angling sideways, far from barcode ...etc

Additionally, verify that `previewView` has the correct width and height (not 0) after applying the changes in this PR. You can confirm this by printing the width and height in the logs.

**Before this PR:** In the BarcodeScanner class, print the width and height before setting the target resolution and observe that both values are 0.

**After this PR:** In the BarcodeScanner class, print the width and height before setting the target resolution and verify that they display the correct values.

Use this website to generate barcodes: https://barcode.tec-it.com/en

### Images/gif

NOTE: The correct barcode value in the video is: `8809573481511`

https://github.com/user-attachments/assets/ff71f1d9-dd63-4887-9cbc-c3d1cde12b02

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/barcodescanner/BarcodeScanner.kt [hunk 1]
```diff
@@ -64,40 +64,42 @@ fun BarcodeScanner(
...
-                        .apply {
-                            setAnalyzer(ContextCompat.getMainExecutor(context), onNewFrame)
-                        }
+                    previewView.post {
+                        val cameraProvider = cameraProviderFuture.get()
+                        val imageAnalysisUseCase = ImageAnalysis.Builder()
+                            .setTargetResolution(
+                                Size(
+                                    previewView.width,
+                                    previewView.height
```
Comment (backwardstruck) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/barcodescanner/BarcodeScanner.kt:73]: Great testing steps. I see this when I log:

```
2024-09-25 14:52:10.247 27818-27818 ItemSelectorDialog      com.woocommerce.android.dev          D  previewView width: 1080
2024-09-25 14:52:10.248 27818-27818 ItemSelectorDialog      com.woocommerce.android.dev          D  previewView height: 2062
```

**Comment by backwardstruck** [comment]
> ...mmerce/android/ui/barcodescanner/BarcodeScanner.kt



> ## [Codecov](https://app.codecov.io/gh/woocommerce/woocommerce-android/pull/12673?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=woocommerce) Report
> Attention: Patch coverage is `0%` with `29 lines` in your changes missing coverage. Please review.
> 
> > Project coverage is 40.58%. Comparing base [(`56404b5`)](https://app.codecov.io/gh/woocommerce/woocommerce-android/commit/56404b537e3b84182e80a54604488e3264d74bcf?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=woocommerce) to head [(`0b55a13`)](https://app.codecov.io/gh/woocommerce/woocommerce-android/commit/0b55a13df5bb13a5bfaf4518a11e4b470b8c82ef?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=woocommerce).
> 
> [Files with missing lines](https://app.codecov.io/gh/woocommerce/woocommerce-android/pull/12673?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=woocommerce)	Patch %	Lines
> [...mmerce/android/ui/barcodescanner/BarcodeScanner.kt](https://app.codecov.io/gh/woocommerce/woocommerce-android/pull/12673?src=pr&el=tree&filepath=WooCommerce%2Fsrc%2Fmain%2Fkotlin%2Fcom%2Fwoocommerce%2Fandroid%2Fui%2Fbarcodescanner%2FBarcodeScanner.kt&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=woocommerce#diff-V29vQ29tbWVyY2Uvc3JjL21haW4va290bGluL2NvbS93b29jb21tZXJjZS9hbmRyb2lkL3VpL2JhcmNvZGVzY2FubmVyL0JhcmNvZGVTY2FubmVyLmt0)	0.00%	[29 Missing ⚠️ ](https://app.codecov.io/gh/woocommerce/woocommerce-android/pull/12673?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=woocommerce)
> Additional details and impacted files
> [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/woocommerce/woocommerce-android/pull/12673?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=woocommerce). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=woocommerce).

I am not sure if there would be value in testing this code. Also hard to test.

**Review by backwardstruck — APPROVED** [review]
Tested in dark mode/light mode, rotated the screen etc using a physical Pixel 6 phone. Definitely noticed it read the barcode successfully more often in this PR.

**Pull request merged**